specifyweb/frontend/js_src/lib/components/TreeView/index.tsx

```diff
@@ -281,7 +281,10 @@ function TreeView<SCHEMA extends AnyTree>({
             onRefresh={(): void => {
               // Force re-load
               setRows(undefined);
-              globalThis.setTimeout(() => setRows(rows), 0);
+              globalThis.setTimeout(() => {
+                setLastFocusedRow(undefined);
```

Comment:
this is only needed if we deleted the focused row right?
in all other cases, this would needlessly cause the focused node to be lost?

edit: looks like the focusedRow is only used by TreeViewActions. so it would cause "Edit" and "Query" button to stop working after you made an edit, until you focus another node

can you confirm if that's what happening? if so, we might want to resolve it. otherwise all good

resolution:
- add an argument to onRefresh like `cause: 'edit' | 'delete'` or something like that

Reply:
Ah, it still works since `focusPath` sets `lastFocusedNode`. That is, even if `lastFocusedNode` is undefined, when the node at `focusPath` gets mounted, it will set the `lastFocusedNode` to itself. Since `focusPath` is still a valid value during other tree operations (synonymize, move, merge), `lastFocusedNode` is not lost (that's a lie, it gets lost immediately after setting to undefined, but it eventually gets set to node at end of `focusPath`)
https://github.com/specify/specify7/blob/e2d9b08a6a61b91723f06b5192494a4ad1948697/specifyweb/frontend/js_src/lib/components/TreeView/Row.tsx#L65-L68

Reply:
^confirmed via testing